datafusion/expr/src/type_coercion/functions.rs

```diff
@@ -414,7 +415,18 @@ fn get_valid_types(
             _ => Ok(vec![vec![]]),
         }
     }
+
     fn array(array_type: &DataType) -> Option<DataType> {
+        match array_type {
```

Comment:
so this says that if the type is a list, keep the type, but if the type is large list / fixed size list then take the field type?

Why doesn't it also take the field type for `List` 🤔 ? (Aka it doesn't make sense to me that List is treated differently than LargeList and FixedSizeList

Reply:
for backwards compat i should keep LargeList so it stays LargeList, will push shortly


> Aka it doesn't make sense to me that List is treated differently than LargeList and FixedSizeList

not my invention, it was like this before.
i think the intention is "converge List, LL and FSL into one type... or maybe two types... to keep UDF impl simpler".

i am not attached to this approach, but i think code may be reliant on that